libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py

```diff
@@ -67,17 +67,47 @@ def run(test, params, env):
 
     device_alias = "ua-" + str(uuid.uuid4())
 
+    def check_device(dev_type, expect_exist=True):
+        """
+        Check the device's availability in vm xml
+
+        :param dev_type: str, device type, like 'watchdog'
+        :param expect_exist: boolean, True if the device is expected to exist
+                            Otherwise, False
+        """
+        domxml = vm_xml.VMXML.new_from_dumpxml(vm_name, options=dump_option)
+        devices = domxml.get_devices(device_type=dev_type)
+        existed = True if devices else False
+        if existed:
+            found = False
+            for one_dev in devices:
+                try:
+                    if one_dev.fetch_attrs()['alias']['name'] == device_alias:
+                        test.log.debug("Found the device with alias '%s'", device_alias)
+                        found = True
+                        break
+                except KeyError as details:
+                    test.log.warning("No key is found: %s", details)
+            existed = True if found else False
+        if existed == expect_exist:
+            return True
+        else:
+            return False
```

Comment:
It can also be updated as below:
```suggestion
        return existed == expect_exist
```

Reply:
Updated